RFC/src/RFC-0120_Consensus.md

```diff
@@ -248,27 +289,73 @@ The [PoW] MUST contain the following:
+\\ \mathrm{ST_i} \text{- the _i_th block's solve time where i <= N}
+\\ \mathrm{ST_{max}} \text{- maximum solve time set in consensus constants}
+\\ \mathrm{T} \text{- target time set in consensus constants}
+\\
+\\ \textit{weighted_solve_time} = \sum_imin(\mathrm{ST_i}, ST_{max})*i
+\\ k = (1+2+3+...+N) * \mathrm{T} / 2
+\\ \textit{difficulty} = \frac{\mathrm{D_{avg}} * k}{\textit{weighted_solve_time}}
+\end{aligned}
+$$
+
+
 ### FTL
+[FTL]: #ftl "Future Time Limit"
 
 The Future Time Limit. This is how far into the future a time is accepted as a valid time. Any time that is more than the FTL is rejected until such a time that it is not less than the FTL.
 The FTL is calculated as (T*N)/20 with T and N defined as:
 T: Target time - This is the ideal time that should pass between blocks which have been mined.
 N: Block window - This is the amount of blocks used when calculating difficulty adjustments.
 
 ### MTP
+[MTP]: #mtp "Median Time Passed"
 
-The Median Time Past. This is the lower limit of a time. Any time that is less than the MTP is rejected.
-THe MTP is calculated as the median timestamp of the previous 11 blocks.
+The Median Time Passed (MTP) is the lower bound calculated by taking the median average timestamp of the 
+last _N_ blocks. Any block with a timestamp that is less than MTP will be rejected.
 
 ### Total accumulated proof of work
 
-This is defined as the total accumulated proof of work done on a single block chain. Because we use two proof of work algorithms which are rated at different difficulties and we would like to weight them as equal we need to compare them. To compare them we use a [Geometric mean](https://en.wikipedia.org/wiki/Geometric_mean). Because we only have two algorithms this is calculated as the ceil of SQRT(accumulated_monero_difficulty*accumulated_blake_difficulty). This number is never transmitted and is simply calculated from the tip of the chain.
+This is defined as the total accumulated proof of work done on the blockchain. Tari uses two _independent_ proof of work algorithms that are
+rated at different difficulties. To compare them, we simply multiply them together into one number 
+(\\( \textit{accumulated_monero_difficulty} * \textit{accumulated_sha_difficulty} \\)). This value is used to compare chain tips to determine
+the strongest chain.
+
+### Transaction Ordering
+[Transaction ordering]: #transaction-ordering "Canonical Transaction Ordering"
+
+The order in which transaction inputs and outputs are added to the Merkle mountain range matters and completely change the
+final Merkle root. TXO ordering is therefore part of consensus. The block MUST be transmitted in canonical ordering.
+The advantage of this approach is that sorting does not need to be done by the whole network (verification of sorting is very cheap).
+
+TXOs are sorted by their Pedersen commitment i.e. `kG + vH`. Specifically, the byte representation of the compressed [ristretto] point 
```

Comment:
Are they sorted in this order? I thought the current implementation sorts them in order of their hash?

Reply:
Yeah the PartialOrd impl for inputs and outputs use the commitment (https://github.com/tari-project/tari/blob/bd017bca0f7f69d3c3c85fbc4385eedbfa37b8b4/base_layer/core/src/transactions/transaction.rs#L528)

I've just looked at kernels and it uses the derive PartialOrd impl

Reply:
If only just because it's convoluted to describe the PartialOrg impl in a spec

Reply:
Interesting, suppose that's what docs are for

Reply:
yeah :) need to discuss if we should change it to use the excess sig